biz.aQute.bndlib/src/aQute/bnd/header/Attrs.java

```diff
@@ -399,7 +399,25 @@ public String toString() {
 
 	public void append(StringBuilder sb) {
 		String del = "";
-		for (Map.Entry<String, String> e : entrySet()) {
+		// Sort entries to ensure consistent ordering of attributes and directives
```

Comment:
If we are going to got this route, then perhaps Attrs.keySet and Attrs.entrySet should be overridden to return items in the desired order. Then everyone benefits and we don't need sorting algorithms which can get out of sync.